src/BundleAnalyzerPlugin.js

```diff
@@ -63,17 +63,27 @@ class BundleAnalyzerPlugin {
 
   generateStatsFile(stats) {
     const statsFilepath = path.resolve(this.compiler.outputPath, this.opts.statsFilename);
-
     mkdir.sync(path.dirname(statsFilepath));
 
-    fs.writeFileSync(
-      statsFilepath,
-      JSON.stringify(stats, null, 2)
-    );
-
-    this.logger.info(
-      `${bold('Webpack Bundle Analyzer')} saved stats file to ${bold(statsFilepath)}`
-    );
+    const options = {
+      promises: 'ignore',
+      buffers: 'ignore',
+      maps: 'ignore',
+      iterables: 'ignore',
+      circular: 'ignore'
+    };
+    return bfj.write(statsFilepath, stats, options)
+      .then(() => {
+        this.logger.info(
+          `${bold('Webpack Bundle Analyzer')} saved stats file to ${bold(statsFilepath)}`
+        );
+      })
+      .catch((error) => {
+        this.logger.error(
+          `${bold('Webpack Bundle Analyzer')} error saving stats file to ${bold(statsFilepath)}.`,
+          JSON.stringify(error, null, '\t')
```

Comment:
Not sure this is the best way to show error. Have you tried that? It prints `{}` for me.

if we want to show stack trace, then let's use `error.stack` instead.

If we only want to show error message, then we can simply inline error object into the template string like this: `${bold('Webpack Bundle Analyzer')} error saving stats file to ${bold(statsFilepath)}: ${error}`.